_source/_posts/2022-01-13-mobile-sso.md

```diff
@@ -0,0 +1,161 @@
+
+This is the recommended approach if your web session may be supporting many different native apps. For example, [FB logout](https://developers.facebook.com/docs/unity/reference/current/FB.Logout/) specifically recommends not to log out of the web session. 
+
+However, if you  require a stronger privacy and security posture, for instance for a banking app, keeping the web session alive may not be an option. 
+
+### Solutions to problem 1
+
+There are two potential approaches to solve problem 1.  Both solutions use browser components that do not show a prompt. Both have the drawback that no cookie can be shared, so SSO will not work. If your app requires SSO, you'll have to find a new way to share login sessions. Fortunately, Okta recently introduced [Native SSO](https://developer.okta.com/docs/guides/configure-native-sso/main/), which allows native apps to implement single sign-on without cookies. See our [blog on SSO between mobile and desktop apps](/blog/2021/11/12/native-sso) for a full example. The following code shows how to remove the prompt, and it assumes your app either does not require SSO or uses Native SSO. 
+
+First, you can use the `prefersEphemeralWebBrowserSession` option for `ASWebAuthenticationSession`. If you are using the [Okta OIDC iOS](https://github.com/okta/okta-oidc-ios) library, you can configure the `noSSO` option as follows:
+
+```swift
+let configuration = OktaOidcConfig(with: {YourOidcConfiguration})
+if #available(iOS 13.0, *) {
+    configuration?.noSSO = true
+}
+```
+
+Under the hook, the `noSSO` option sets the `prefersEphemeralWebBrowserSession` flag. Note that this flag is only available in iOS 13 and above. 
+
+Second, if you desire to support older iOS versions, you could use `SFSafariViewController` as the browser to present the login session. The following demonstrates how to launch `SFSafariViewController` if you are using the [AppAuth iOS](https://github.com/openid/AppAuth-iOS) library. 
+
+AppAuth iOS supports a concept of *external user agent*, where you can use any browser to present the sign-in webpage. By default, if you invoke `OIDAuthState.authStateByPresentingAuthorizationRequest:presentingViewController:callback:`, it will invoke a default user agent. But you can plug in any external user agent, as long as it conforms to the `OIDExternalUserAgent` protocol. There is already an implementation of using [`SFSafariViewController` as an external agent](https://gist.github.com/ugenlik/2a543f351e9b9425800b48266760dc85). You can download it and plug it into your project. Then, when you invoke the browser, call 
+`OIDAuthState.authStateByPresentingAuthorizationRequest:presentingViewController:callback:`, and pass the external user agent `OIDExternalUserAgentIOSSafariViewController`, as shown in the following snippet. 
```

Comment:
These single backtick blocks are too long and get chopped and cut off on smaller screens.

Consider rewording to wrap in a triple backtick, or otherwise introduce the fqn in the example below?
Or maybe reword to something like:

When the `OIDAuthState` callback is invoked...

(i know it's a bit of a hackie work around)

Reply:
Please review the new version breaking the long method into triple quote.